src/pubmetric/api_controller.py

```diff
@@ -0,0 +1,77 @@
+from fastapi import FastAPI, UploadFile, File
+from pydantic import BaseModel
+from typing import Dict
+import tempfile
+import os 
+import json
+from datetime import datetime
+
+from pubmetric.metrics import *
+from pubmetric.workflow import parse_cwl_workflows
+from pubmetric.network import create_citation_network
+
+app = FastAPI()
+
+class ScoreResponse(BaseModel):
+    workflow_scores: Dict
+
+class GraphRequest(BaseModel):
+    topic_id: str
+    test_size: str
+
+@app.post("/score_workflows/", response_model=ScoreResponse)
+async def score_workflows(file: UploadFile = File(None)):
+    path_to_data = "out_202407041439" # where does one store things later?  
```

Comment:
the file should probably have a descriptive name (that is used as a constant) and be locally stored